references/plugins/cluster.go

```diff
@@ -290,11 +301,12 @@ func SyncDefinitionToLocal(ctx context.Context, c common.Args, localDefinitionDi
 		foundCapability = true
 	}
 	if foundCapability {
-		template, err := HandleDefinition(capabilityName, traitDef.Spec.Reference.Name,
+		template, err = HandleDefinition(capabilityName, traitDef.Spec.Reference.Name,
 			traitDef.Annotations, traitDef.Spec.Extension, types.TypeTrait, nil, traitDef.Spec.Schematic)
-		if err == nil {
-			return &template, nil
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to handle TraitDefinition")
 		}
+		return &template, nil
 	}
 	return nil, fmt.Errorf("%s is not a valid workload type or trait", capabilityName)
```

Comment:
After the refactor, will here still have any errors? I hope we can make the code clean.

Reply:
Yes, there would be a `capability xxx cloud not be found in the current env`. Let me make it clear.